fleetctl/journal.go

```diff
@@ -17,39 +17,57 @@ package main
 import (
 	"strconv"
 
-	"github.com/codegangsta/cli"
+	"github.com/spf13/cobra"
 
 	"github.com/coreos/fleet/client"
 	"github.com/coreos/fleet/job"
 )
 
-func NewJournalCommand() cli.Command {
-	return cli.Command{
-		Name:      "journal",
-		Usage:     "Print the journal of a unit in the cluster to stdout",
-		ArgsUsage: "[--lines=N] [--ssh-port=N] [-f|--follow] [--output=STRING] <unit>",
-		Action:    makeActionWrapper(runJournal),
-		Description: `Outputs the journal of a unit by connecting to the machine that the unit occupies. 
+var (
+	flagLines  int
+	flagFollow bool
+	flagSudo   bool
+	flagOutput string
+)
+
+var cmdJournal = &cobra.Command{
+	Use:   "journal [--lines=N] [--ssh-port=N] [-f|--follow] [--output=STRING] <unit>",
+	Short: "Print the journal of a unit in the cluster to stdout",
+	Long: `Outputs the journal of a unit by connecting to the machine that the unit occupies.
 
 Read the last 10 lines:
-       fleetctl journal foo.service
+fleetctl journal foo.service
 
 Read the last 100 lines:
-       fleetctl journal --lines 100 foo.service
+fleetctl journal --lines 100 foo.service
 
 This command does not work with global units.`,
-		Flags: []cli.Flag{
-			cli.IntFlag{Name: "lines", Value: 10, Usage: "Number of recent log lines to return"},
-			cli.BoolFlag{Name: "follow, f", Usage: "Continuously print new entries as they are appended to the journal."},
-			cli.IntFlag{Name: "ssh-port", Value: 22, Usage: "Connect to remote hosts over SSH using this TCP port"},
-			cli.BoolFlag{Name: "sudo", Usage: "Execute journal command with sudo"},
-			cli.StringFlag{Name: "output", Value: "short", Usage: "Output mode. This will be passed unaltered to journalctl on the remote host, and hence supports the same modes as that command."},
-		},
-	}
+	Run: runWrapper(runJournal),
+}
+
+func init() {
+	cmdFleet.AddCommand(cmdJournal)
+
+	cmdJournal.Flags().IntVar(&flagLines, "lines", 10, "Number of recent log lines to return")
+	cmdJournal.Flags().BoolVar(&flagFollow, "follow", false, "Continuously print new entries as they are appended to the journal.")
+	cmdJournal.Flags().BoolVar(&flagFollow, "f", false, "Shorthand for --follow")
+	cmdJournal.Flags().IntVar(&sharedFlags.SSHPort, "ssh-port", 22, "Connect to remote hosts over SSH using this TCP port")
+	cmdJournal.Flags().BoolVar(&flagSudo, "sudo", false, "Execute journal command with sudo")
+	cmdJournal.Flags().StringVar(&flagOutput, "output", "short", "Output mode. This will be passed unaltered to journalctl on the remote host, and hence supports the same modes as that command.")
+}
+
+// GetJournalCommand returns the cobra command for journal
+func GetJournalCommand() *cobra.Command {
+	return cmdJournal
 }
 
-func runJournal(c *cli.Context, cAPI client.API) (exit int) {
-	args := c.Args()
+func runJournal(cCmd *cobra.Command, args []string, fakeAPI client.API) (exit int) {
+	if fakeAPI == nil {
```

Comment:
this is an unfortunate duplication pasted everywhere - can it go within the wrapper as it used to? (e.g. formerly makeActionWrapper)